pilot/pkg/bootstrap/server.go

```diff
@@ -1046,10 +1045,11 @@ func (s *Server) maybeCreateCA(caOpts *caOptions) error {
 			}
 		}
 		// May return nil, if the CA is missing required configs - This is not an error.
-
-		// TODO: Issue #27606 If External CA is configured, use that to sign DNS Certs as well. IstioCA need not be initialized
-		if s.CA, err = s.createIstioCA(corev1, caOpts); err != nil {
-			return fmt.Errorf("failed to create CA: %v", err)
+		noCA := features.PilotCertProvider == constants.CertProviderKubernetes && caOpts.ExternalCASigner != ""
```

Comment:
I would suggest having this logic when registering/running the CA grpc service. The logic is getting a little too complex and we should avoid  having this in too many places. Could we have this in a function?

Reply:
sure, move this into disableCAServer()